modules/misc/qt.nix

```diff
@@ -44,19 +44,63 @@ in {
           </variablelist>
         '';
       };
+
+      style = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "adwaita";
+        relatedPackages = [ "adwaita-qt" [ "libsForQt5" "qtstyleplugins" ] ];
+        description = ''
+          Selects the style to use for Qt5 applications.</para>
+          <para>The options are
+          <variablelist>
+            <varlistentry>
+              <term><literal>adwaita</literal></term>
+              <term><literal>adwaita-dark</literal></term>
+              <listitem><para>Use Adwaita Qt style with
+                <link xlink:href="https://github.com/FedoraQt/adwaita-qt">adwaita</link>
+              </para></listitem>
+            </varlistentry>
+            <varlistentry>
+              <term><literal>cleanlooks</literal></term>
+              <term><literal>gtk2</literal></term>
+              <term><literal>motif</literal></term>
+              <term><literal>plastique</literal></term>
+              <listitem><para>Use styles from
+                <link xlink:href="https://github.com/qt/qtstyleplugins">qtstyleplugins</link>
+              </para></listitem>
+            </varlistentry>
+          </variablelist>
+        '';
+      };
     };
   };
 
   config = mkIf (cfg.enable && cfg.platformTheme != null) {
-    home.sessionVariables.QT_QPA_PLATFORMTHEME =
-      if cfg.platformTheme == "gnome" then "gnome" else "gtk2";
+    assertions = [{
+      assertion = (cfg.platformTheme == "gnome") -> (cfg.style != null);
+      message = ''
+        `qt.platformTheme` "gnome" must have `qt.style` set to a theme that
+        supports both Qt and Gtk, for example "adwaita" or "adwaita-dark".
+      '';
+    }];
+
+    # Necessary because home.sessionVariables is of types.attrs
+    home.sessionVariables = (filterAttrs (n: v: v != null) {
+      QT_QPA_PLATFORMTHEME =
+        if cfg.platformTheme == "gnome" then "gnome" else "gtk2";
+      QT_STYLE_OVERRIDE = cfg.style;
+    });
 
     home.packages = if cfg.platformTheme == "gnome" then
       [ pkgs.qgnomeplatform ]
+      ++ lib.optionals (builtins.elem cfg.style [ "adwaita" "adwaita-dark" ])
+      [ pkgs.adwaita-qt ]
```

Comment:
Now that the user can pass any theme here, this seems wrong. I am thinking maybe adding something like `qt.style.package`, but I don't know if this makes sense considering the user can simply add the package to `home.packages`.

Maybe just documenting this somewhere that you need to add the theme package to `home.packages` :thinking: ?

Reply:
@berbiche Any comment about this :point_up: ?

Reply:
`qt.style.package` is a good idea :+1:

Reply:
@berbiche Done, can you review again?

Reply:
@thiagokokada can you remove the empty file `tests/modules/misc/qt/qt-platform-theme-expected`? It still doesn't seem to be used

Reply:
@berbiche Done.